app/controllers/assign_mentor.rb

```diff
@@ -0,0 +1,10 @@
+class AssignMentor
+  def self.assign(mentee, course_id)
+    available_mentors = Mentor.where(available: true)
+    assigned_mentor = available_mentors.find do |mentor|
+      mentor.courses.where(id: course_id)
+    end
```

Comment:
This looks like a possible n+1 query problem. If you have `n` number of available mentors, then you will generate `n` queries.

Reply:
Also, is it possible to have multiple mentors teaching the same course? Do you just find the first available mentor to assign in that case?

Reply:
Yes, it is possible to have the first and therefore we pick the first available mentor for that course